extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3StorageDruidModule.java

```diff
@@ -182,4 +187,20 @@ public Supplier<ServerSideEncryptingAmazonS3> getAmazonS3ClientSupplier(
   {
     return Suppliers.memoize(serverSideEncryptingAmazonS3Builder::build);
   }
+
+  @Provides
+  @LazySingleton
+  @Named("S3UploadThreadPool")
+  public ExecutorService getUploadPoolExecutorService(ScheduledExecutorFactory scheduledExecutorFactory, RuntimeInfo runtimeInfo)
+  {
+    int poolSize = Math.max(4, 3 * runtimeInfo.getAvailableProcessors());
```

Comment:
What's the rationale behind this? It seems super weird that we are allocating more threads to uploading than the number of processors. Check the basic cluster tuning section: https://druid.apache.org/docs/latest/operations/basic-cluster-tuning/#processing-threads-and-buffers. 
The number of processing threads is limited to the number of processors to avoid over utilisation. 

In any case, this value should be extracted somewhere and a Javadoc should be present behind this logic.

Reply:
@LakshSingla I made this change as per: https://github.com/apache/druid/pull/16481#discussion_r1609309586

cc: @cryptoe Can you chime in? Appreciate your thoughts here, thanks!

Reply:
I've changed it to `Math.max(4, runtimeInfo.getAvailableProcessors());` as Kashif also suggested to reduce this. Hope that works!